docs/scripts/generate-hardware-matrix.py

```diff
@@ -47,9 +48,7 @@ def main() -> None:
         if hw_requirement == 'arch':
             continue
 
-        matrix[hw_requirement] = {
-            method: (False, int) for method in known_methods
-            }
+        matrix[hw_requirement] = {method: (False, int) for method in known_methods}
```

Comment:
For this change, does the following work?
```suggestion
        matrix[hw_requirement] = {
            method: (False, int)
            for method in known_methods,
            }
```

Reply:
I'm not sure how to convince `ruff` to keep your suggestion (which I like as well). It always converts it into one-line.

Reply:
@happz this part is also a bit tricky and you should check out. Without manual intervention, `ruff format` will make this a single-line. We can however prevent it manually by adding `# fmt: skip` (otherwise it decides by line-length alone). IMO, for parts of the code where we initialize a `dict` with this for construct, it makes sense to have a special case to force it with `# fmt: skip` like in the suggestion above. Wdyt?

Reply:
A single line is fine with me. I like the two lines better though, but I'm not sure it's worth the effort to add `# fmt: skip` everywhere we initialize a dictionary, it might become too much work. So, no preference for me, I can live with both.

Reply:
I mean specifically for those that include a `for` loop like the example above because it is a special construct that cannot be combined with normal dict initializations. And mostly to decide on if we _allow_ the exception if the author/reviewer catches something that they would prefer, and if we don't catch them all, that's fine, probably it would be amended with new releases of `ruff`

Reply:
I'd be also good with the single line. When the expression is long enough `ruff-format` nicely separates the `for` and `if` on separate lines and it looks nicely readable.